### PR TITLE
#51: Fixed test button in job configuration section

### PR DESCRIPTION
**Pull request by peergum**

### CODE REVIEW

**Comment by peergum** [comment]
This fixes the issues with the Test button in the job configuration section. Tests work using specific values from job section or default values from general configuration.

**Comment by samrocketman** [comment]
Adding the Travis CI file doesn't make sense.  I don't have permissions to update the configuration settings of this GitHub project (to even add Travis CI as an integration).  Plus, verify builds are performed by hosted Jenkins.

Can you amend this pull request so that Travis CI commits are not there?  Might I suggest the following to update it.

```bash
#add the upstream slack plugin repository as a remote
git remote add upstream git@github.com:jenkinsci/slack-plugin.git
git fetch upstream
#rename master branch to old_master
git branch -m old_master
#check out a new master
git checkout upstream/master -b master
#cherry pick commit #51: Fixed test button in job notification section
git cherry-pick 3312ebc
#update the pull request
git push origin master --force
```
Force pushing the branch automatically updates the Pull Request in GitHub.

File: src/main/java/jenkins/plugins/slack/SlackService.java [hunk 1]
```diff
@@ -3,5 +3,5 @@
 public interface SlackService {
     void publish(String message);
 
-    void publish(String message, String color);
```
Comment (samrocketman) [inline src/main/java/jenkins/plugins/slack/SlackService.java]: This should remain indented.

File: README.md [hunk 2]
```diff
@@ -1,5 +1,4 @@
-# Slack plugin for Jenkins
-
+# Slack plugin for Jenkins [![Build Status](https://travis-ci.org/jenkinsci/slack-plugin.svg?branch=master)](https://travis-ci.org/jenkinsci/slack-plugin)
```
Comment (samrocketman) [inline README.md]: There's no Travis CI build so no need to provide a link in the README.  Can you amend this commit?

Comment (samrocketman) [inline README.md]: Assuming you still have the `upstream` remote configured, might I suggest the following.

```bash
git checkout upstream -- README.md
git add README.md
git commit --amend
git push origin --force
```

**Comment by peergum** [comment]
all changes done and rebased.

**Comment by peergum** [comment]
Fixed as suggested by Trevor Bernard (using .equals), merged the build_status in and rebased.

**Comment by samrocketman** [comment]
This still does not work.  I can't get it to fail.  It 100% succeeds all the time even when I enter incorrect information.  Try to get the test connection button to fail.

**Comment by samrocketman** [comment]
I was able to get `Client error : null` when I tested with no settings defined.  That is an improvement.  However, when I put in a wrong domain it still says it succeeded even when the stack track on the CLI says it failed.

**Comment by peergum** [comment]
I didn't correct the failure part so far... only reused the general settings where local settings were missing... I'll try to check that asap.

**Comment by samrocketman** [comment]
If you could add commits to this PR to address that then that'd be great.  If you'd like that to be a separate pull request then let me know.  I'll merge this one and await your other pull request addressing the issue I mentioned.

**Comment by samrocketman** [comment]
I'm going to hold off merging this until we can fully fix the Test Connection button.  That being said I'll delay the next release until this fix is in.

**Comment by samrocketman** [comment]
I guess since you're going your own path with this I'll close the open request.  I encourage you to reconsider contributing to the plugin.  If you feel the plugin is going the wrong direction you're welcome to open pull requests to shepherd it.  In any case, you're always welcome.

**Comment by samrocketman** [comment]
I'm also open to change of the process in which this plugin is developed.  As a maintainer, it's difficult to tell what should be merged or shouldn't so perhaps a minimum incubation period for pull requests will be implemented as a result of this interaction.  I meant no harm when I originally stepped on your toes.  Transparency and compatibility is certainly a goal along with improvement and innovation.

**Pull request closed without merging**